Update crayons Modal to add sheet variant (#17576)
forem/forem: app/javascript/crayons/Modal/__stories__/modals.mdx

- Small - 480px - for displaying small amount of data/text.
- Medium - 640px (this is default one) - you will probably want to use this one in most cases.
- Large - 768px - use for long and/or complex content.

## Sheets

Sheets are a special variant of the modal, which will occupy the full height of the screen. Sheets may also be anchored to the left or right side of the page by using the `sheetAlign` prop.

## Prompts

> Except classic Modals you can also use `prompt` style (prop) to apply _special styling_.

Reviewer: Quick, non-blocking suggestion for clarity purposes:
```suggestion
Classic modals excluded, you can also use the `prompt` style (prop) to apply _special styling_.
```

### Centered prompt

Some prompts may visually benefit from centering the content.

> Leave the decision to designers but keep it mind this is possible with `centered` prop

Reviewer: Quick typo fix!:
```suggestion
Leave the decision to designers but keep in mind this is possible with `centered` prop
```

(and it will only work in combination with `prompt`).

## Backdrop

> By default Modals come with dark backdrop (a layer overlaying underlaying content).

Reviewer: Quick, non-blocking suggestion for clarity purposes:
```suggestion
By default, Modals come with a dark backdrop (a layer overlaying underlaying content).
```

Author: Thanks for these fixes! 🎉
